lib/src/collection/treeset.dart

```diff
@@ -436,6 +436,8 @@ class AvlTreeSet<V> extends TreeSet<V> {
   }
 
   bool remove(Object item) {
+    if (item is! V) return false;
+
     AvlNode<V> x = _getNode(item as V);
```

Comment:
Shame that the analyzer doesn't infer that... FYI filed https://github.com/dart-lang/sdk/issues/26195